backend-api/src/functions/common/logging/setupLogger.ts

```diff
@@ -1,8 +1,43 @@
 import { Context } from "aws-lambda";
 import { logger } from "./logger";
 
-export const setupLogger = (context: Context) => {
+export const setupLogger = (
+  context: Context,
+  userAgent: string | undefined,
+) => {
   logger.resetKeys();
   logger.addContext(context);
-  logger.appendKeys({ functionVersion: context.functionVersion });
+  logger.appendKeys({
+    functionVersion: context.functionVersion,
+    userAgent: buildUserAgent(userAgent),
+  });
+};
+
+export const buildUserAgent = (
+  userAgent: string | undefined,
+): object | undefined => {
+  if (userAgent === undefined || userAgent === "") {
+    return { userAgentHeader: "", deviceType: "unknown" };
+  }
+
+  const anyPartHasPrefix = (parts: string[], prefix: string): boolean => {
+    return parts.some(function (part) {
+      return part.startsWith(prefix);
+    });
+  };
+
+  const userAgentParts: string[] = userAgent.split(" ");
+
+  const isAndroid =
+    anyPartHasPrefix(userAgentParts, "Android/") ||
+    anyPartHasPrefix(userAgentParts, "Dalvik/");
+  const isIphone =
+    anyPartHasPrefix(userAgentParts, "iOS/") ||
+    anyPartHasPrefix(userAgentParts, "Darwin/");
+
+  const deviceTypes: string[] = ["unknown", "Android", "iPhone", "lol"];
+  return {
+    userAgentHeader: userAgent,
```

Comment:
note: if you decided to action [this comment](https://github.com/govuk-one-login/mobile-id-check-async/pull/858/files#r2401783260), you can update this to


```suggestion
    userAgentHeader,
```